Sources/BraveWallet/Crypto/Stores/CryptoStore.swift

```diff
@@ -386,12 +386,34 @@ public class CryptoStore: ObservableObject {
     return pendingRequest != nil
   }
 
-  func handleWebpageRequestResponse(_ response: WebpageRequestResponse) {
+  struct AddNetworkCompletion {
+    let chainId: String
+    let completion: (_ error: String?) -> Void
+  }
+  // Helper to store the completion block of an Add Network request.
+  // The completion closure is handled in `onAddEthereumChainRequestCompleted`
+  // when we determine if the chain was added successfully or not.
+  var addNetworkWebpageRequestCompletion: AddNetworkCompletion?
```

Comment:
shouldn't this be a list? what if while adding network 1, im adding network 2 that also takes 4 min ?

Reply:
These add chain requests are stored in a [flat_map](https://chromium.googlesource.com/chromium/src/base/+/master/containers/flat_map.h#32) in [brave-core](https://github.com/brave/brave-core/blob/master/components/brave_wallet/browser/json_rpc_service.h#L607-L609) which [behaves a bit more like an array for us when we fetch these requests](https://github.com/brave/brave-core/blob/master/components/brave_wallet/browser/json_rpc_service.h#L607-L609) so while I don't think this is currently an issue, I think it's probably best to update here to treat these requests as if they were stored a dictionary; I'll try and update this to be safe.